# Using Large Language Models (LLMs) for Bioinformatics Research

<img src="https://github.com/lynnlangit/TeamTeri/blob/master/Images/llm-research.png" width=500 align=center>

## Introduction to using LLMs in Bioinformatics  

* **Brief explanation of LLMs and their potential for transforming bioinformatics:**
  - [Harnessing the Power of Large Language Models for Bioinformatics](https://medium.com/@andrew_johnson_4/unveiling-the-potential-of-large-language-models-in-bioinformatics-76710eb0b38a)
* **Discussion of key challenges faced by traditional bioinformatics approaches:**
  - [Overcoming the Limitations of Traditional Bioinformatics Techniques](https://towardsdatascience.com/overcoming-the-limitations-of-large-language-models-9d4e92ad9823)
* **Outline of the advantages LLMs offer:**
  - [Large Language Models in Molecular Biology](https://towardsdatascience.com/large-language-models-in-molecular-biology-9eb6b65d8a30)
  - [Take the DNA Delorean: The Promise of Large Language Models in Genomics](https://www.bio-itworld.com/news/2023/01/27/take-the-dna-delorean-the-promise-of-large-language-models-in-genomics)
* **The next frontier for LLMs is biology:**
  - [Why Bioinformatics Needs Large Language Models](https://www.forbes.com/sites/robtoews/2023/07/16/the-next-frontier-for-large-language-models-is-biology/?sh=7a7084506f05)
* **LLM Tutorials:**
  - :octocat: [LLM Tutorials from Argonne Labs](https://github.com/langitlynn/llm-workshop)

## Key Applications of LLMs in Bioinformatics

<img src="https://github.com/lynnlangit/TeamTeri/blob/master/Images/llm-bio.png" width=300 align=right>

* **Sequence Analysis:**
  - [Large models for genomics](https://www.nature.com/articles/s41592-023-02105-5)
  - [ProtHyena: A fast and efficient foundation protein language model at single amino acid Resolution](https://www.biorxiv.org/content/10.1101/2024.01.18.576206v1)
* **Protein Structure Prediction:**
  - [Protein Structure Prediction with AI: AlphaFold](https://www.nature.com/articles/s41586-021-03819-2%3C)
  - [When Protein Structure Embedding Meets Large Language Models](https://pubmed.ncbi.nlm.nih.gov/38254915/)
  - [OpenFold Biotech AI Research Consortium releases SoloSeq and Multimer, an integrated protein Large Language Model with 3D structure generation](https://www.businesswire.com/news/home/20240219658831/en/OpenFold-Biotech-AI-Research-Consortium-releases-SoloSeq-and-Multimer-an-integrated-protein-Large-Language-Model-with-3D-structure-generation)
* **Drug Discovery:**
  - [Large Language Models may speed Drug Discovery](https://www.technologyreview.com/2023/08/22/1076802/large-language-models-may-speed-drug-discovery/)
* **Gene Expression Analysis:**  
  - [Custom GPT for gene analysis](https://chat.openai.com/g/g-HvvFt9ohw-genegpt)
  - [GeneGPT: Augmenting Large Language Models with Domain Tools for Improved Access to Biomedical Information](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10153281/)
  - [GENEPT: A SIMPLE BUT HARD-TO-BEAT FOUNDATION MODEL FOR GENES AND CELLS BUILT FROM CHATGPT](https://www.biorxiv.org/content/10.1101/2023.10.16.562533v1)
* **Digi Image Path Analysis:**
  - [A visualâ€“language foundation model for pathology image analysis using medical Twitter](https://www.nature.com/articles/s41591-023-02504-3)
  - [Quilt-1M: One Million Image-Text Pairs for Histopathology](https://arxiv.org/abs/2306.11207)
  - [Text-guided Foundation Model Adaptation for Pathological Image Classification](https://arxiv.org/abs/2307.14901)
  - [Transformer with convolution and graph-node co-embedding: An accurate and interpretable vision backbone for predicting gene expressions from local histopathological image](https://www.sciencedirect.com/science/article/pii/S1361841523003006)
  
## Challenges and Limitations

* **Data Quality and Bias:**
  - [THE ROLE OF NATURAL LANGUAGE PROCESSING IN HEALTHCARE](https://roundtable.datascience.salon/the-role-of-natural-language-processing-in-healthcare)
* **Computational Requirements:**
  - [Leveraging Fine-Tuned Large Language Models in Bioinformatics: A Research Perspective](https://www.qeios.com/read/WE7UMN.2)
* **Interpretability:**
  - [XAI meets Biology: A Comprehensive Review of Explainable AI in Bioinformatics Applications](https://arxiv.org/html/2312.06082v1)
* **Benchmarking:**
  - [Bioinfo-Bench: A Simple Benchmark Framework for LLM Bioinformatics Skills Evaluation](https://www.biorxiv.org/content/10.1101/2023.10.18.563023v1)

## Summary and Future Directions
* **Potential new application areas:**
  - [Precision Health in the Age of Large Language Models](https://dl.acm.org/doi/10.1145/3580305.3599568)
* **Advancements in LLM capabilities:** 
  - [Applications of transformer-based language models in bioinformatics: a survey](https://academic.oup.com/bioinformaticsadvances/article/3/1/vbad001/6984737)
* **Summarize the transformative potential of LLMs in bioinformatics:**
  - [The rise of large language models (LLMs)](https://omicstutorials.com/the-rise-of-large-language-models-llms/)
* **Reiterate key takeaways and the exciting future for this field:**
   - [Large Language Models (LLMs) in Science and Biology: Advancements in Text Generation and Creative AI](https://omicstutorials.com/large-language-models-llms-in-science-and-biology-advancements-in-text-generation-and-creative-ai/)

-----


